sequence_processing_pipeline/Commands.py

```diff
@@ -130,3 +130,58 @@ def demux(id_map, fp, out_d, task, maxtask):
     for d in openfps.values():
         for f in d.values():
             f.close()
+
+
+def annotate_filtered_fastq(original_path, stripped_path, output_path):
+    """
+    Annotates a FASTQ file w/missing descriptions using original file.
+    :param original_path: A path to the original, unfiltered file.
+    :param stripped_path: A path to the filtered file.
+    :param output_path: A path for the output file.
+    :return: None
+    """
+    mapping = {}
+
+    # Per FASTQ specification: https://en.wikipedia.org/wiki/FASTQ_format
+    # lines beginning with '@' contain the sequence's identifier and an
+    # optional description or 'metadata'.
+
+    # minimap2 is a tool used to perform host filtering on fastq files.
+    # The preferred output for this program is SAM format:
+    # https://en.wikipedia.org/wiki/SAM_(file_format)
+
+    # SAM format cannot preserve the optional metadata field and hence the
+    # metadata will disappear after using minimap2 to filter a file and
+    # samtools to convert the file from SAM back to FASTQ format.
+
+    # It is straightforward to process even large fastq files line by line
+    # sequence identifier lines are unique and easily identified as they are
+    # the only lines in a FASTQ file beginning with '@'. It is thus straight-
+    # forward to process even large files to build a mapping between
+    # sequence identifiers and optional metadata fields.
+    with open(original_path, 'r') as f:
+        for line in f:
+            if line.startswith('@'):
+                line = line.strip().split()
+                if len(line) != 2:
+                    raise ValueError(f"'{original_path}' does not appear to "
```

Comment:
Fine to raise a `ValueError` here, but is a per-read metadata guaranteed? If not, might be worth more gracefully handling this to avoid halting.

Reply:
The FASTQ spec doesn't say every sequence identifier must contain optional descriptions or none of them do, although this appears to be our current situation. My thinking was for the moment it's better to raise an error and let the user know that one of the sequence identifiers doesn't contain a description, rather than letting something silently fail downstream. @wasade is perhaps the person to ask.

Reply:
All reads from tellread will have the BX:Z tag or whatever it is. In general, it is not safe to assume a description will lack whitespace though. It is correct to except if the comment is not present for tellread as we expect it to be there

Reply:
gotcha. I'll change the split to split on the first space only.